### PR TITLE
Fix show instrumented dir

### PR DESCRIPTION
**Pull request by herzogp**
[ENG-291](https://linear.app/antithesis/issue/ENG-291/go-sdk-instrument-and-catalog-for-test-and-protobuf-sources) Go  SDK should include intrumentation and cataloging for 'test' and 'protobuf' sources.  Now these files are instrumented by default, but can be ignored with instrumentor CLI options

[ONCALL-25](https://linear.app/antithesis/issue/ONCALL-25/go-instrumentation-go-mod-tidy-fails)
Go mod tidy fails after instrumenting - due to using `cp -n -r` from busybox rather than using a fully working `cp --noclobber -r`.  Now using a copy/no-replace directory tree mechanism implemented entirely in Go, rather than relying upon `cp` command.

[ENG-182](https://linear.app/antithesis/issue/ENG-182/when-using-the-go-instrumentor-by-default-it-uses-latest-sdk-and-does) When using the go instrumentor by default it uses latest SDK and does not respect the SDK version of the instrumentor being used on customer code.  Now retrieves the eversion number of the instrumentor, and uses this as the default.  The default is overrideable with a CLI option.

### CODE REVIEW

File: tools/antithesis-go-instrumentor/common/files.go [hunk 1]
```diff
@@ -45,22 +47,175 @@ func WriteTextFile(text, file_name string) (err error) {
...
+	}
+
+	// Make sure the 'to' path is a directory that exists
+	if err = EnsureDirExists(to); err != nil {
+		return
+	}
+
+	// Read the 'from' directory (and close its *File)
+	if fromDirEntries, err = fromDir.ReadDir(0); err != nil {
+		fromDir.Close()
```
Comment (herzogp) [inline tools/antithesis-go-instrumentor/common/files.go:159]: Please comment that errors on .Close() are being ignored

File: tools/antithesis-go-instrumentor/common/files.go [hunk 2]
```diff
@@ -45,22 +47,175 @@ func WriteTextFile(text, file_name string) (err error) {
...
+	if !dirInfo.IsDir() {
+		dir.Close()
+		dir = nil
+		err = fmt.Errorf("%q is not a directory", dirName)
+		return
+	}
+	return
+}
+
+func quietClose(fps ...*os.File) {
```
Comment (herzogp) [inline tools/antithesis-go-instrumentor/common/files.go:87]: Comment on the fact that quietclose will not error on .Close() failure

File: tools/antithesis-go-instrumentor/common/files.go [hunk 3]
```diff
@@ -45,22 +47,175 @@ func WriteTextFile(text, file_name string) (err error) {
...
+	for _, fp := range fps {
+		if fp != nil {
+			fp.Close()
+		}
+	}
+}
+
+func CopyFile(from, to string) (err error) {
```
Comment (herzogp) [inline tools/antithesis-go-instrumentor/common/files.go:95]: Add a comment what this function does

File: tools/antithesis-go-instrumentor/common/files.go [hunk 4]
```diff
@@ -45,22 +47,175 @@ func WriteTextFile(text, file_name string) (err error) {
...
+func OpenExistingDir(dirName string) (dir *os.File, err error) {
+	var dirInfo os.FileInfo
+
+	if dir, err = os.Open(dirName); err != nil {
+		dir = nil
+		return
+	}
+
+	if dirInfo, err = dir.Stat(); err != nil {
+		dir.Close()
```
Comment (herzogp) [inline tools/antithesis-go-instrumentor/common/files.go:73]: should be quietclose() here, and elsewhere that the return value from . Close() is being ignored.

File: tools/antithesis-go-instrumentor/common/files.go [hunk 5]
```diff
@@ -36,31 +38,200 @@ func WriteTextFile(text, file_name string) (err error) {
...
+		return
+	}
+	err = os.Chmod(to, fileInfo.Mode())
+	return
+}
+
+func CopyRecursiveDir(from, to string) (err error) {
+
+	var fromDir *os.File
+	// var toDir *os.File
```
Comment (JRMurr) [inline tools/antithesis-go-instrumentor/common/files.go:157]:
```suggestion
```

File: tools/antithesis-go-instrumentor/common/files.go [hunk 6]
```diff
@@ -36,31 +38,200 @@ func WriteTextFile(text, file_name string) (err error) {
...
+		quietClose(dir)
+		dir = nil
+		return
+	}
+
+	if !dirInfo.IsDir() {
+		quietClose(dir)
+		dir = nil
+		err = fmt.Errorf("%q is not a directory", dirName)
+		// return
```
Comment (JRMurr) [inline tools/antithesis-go-instrumentor/common/files.go:90]:
```suggestion
```

File: tools/antithesis-go-instrumentor/cmd/cmd_files.go [hunk 7]
```diff
@@ -172,8 +180,16 @@ func (cfx *CommandFiles) WrapUp() {
 		}
 	}
 
-	common.CopyRecursiveNoClobber(cfx.inputDirectory, cfx.customerDirectory)
-	cfx.logWriter.Printf("All other files copied unmodified from %s to %s", cfx.inputDirectory, cfx.customerDirectory)
+	// common.CopyRecursiveNoClobber(cfx.inputDirectory, cfx.customerDirectory)
```
Comment (JRMurr) [inline tools/antithesis-go-instrumentor/cmd/cmd_files.go:183]:
```suggestion
```

**Review by JRMurr — APPROVED** [review]
Some small nits to remove commented out code but otherwise lgtm

File: tools/antithesis-go-instrumentor/cmd/cmd_files.go (hunk 7 above)
Comment (herzogp) [reply tools/antithesis-go-instrumentor/cmd/cmd_files.go:183]: Thank you

File: tools/antithesis-go-instrumentor/common/files.go (hunk 6 above)
Comment (herzogp) [reply tools/antithesis-go-instrumentor/common/files.go:90]: Thank you

File: tools/antithesis-go-instrumentor/common/files.go (hunk 5 above)
Comment (herzogp) [reply tools/antithesis-go-instrumentor/common/files.go:157]: Thank you - all suggestions adopted/fixed.

**Pull request merged**